crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -680,7 +683,26 @@ impl<'t> TextEdit<'t> {
                     }
 
                     if text.is_mutable() {
-                        paint_cursor(&painter, ui.visuals(), primary_cursor_rect);
+                        // is_blink: Stays displayed when the cursor is moving
+                        let blink = ui.visuals().text_cursor_blink;
+                        let is_blink = blink && (save_ccursor_range == state.cursor.char_range());
+                        let is_cursor_visible = paint_cursor(
+                            &painter,
+                            ui.visuals(),
+                            primary_cursor_rect,
+                            i_time,
+                            is_blink,
+                        );
+                        if is_blink {
+                            if is_cursor_visible {
+                                ui.ctx()
+                                    .request_repaint_after(std::time::Duration::from_millis(700));
+                            }
+                            if !is_cursor_visible {
+                                ui.ctx()
+                                    .request_repaint_after(std::time::Duration::from_millis(300));
```

Comment:
Make the cursor painter return the number of seconds until the next time the cursor should change from visible to invisible

Reply:
> Make the cursor painter return the number of seconds until the next time the cursor should change from visible to invisible

Would it be good to be able to set the time the cursor is on and off in `ctx.settings_ui()`?